sky/backends/backend_utils.py

```diff
@@ -1629,21 +1626,23 @@ def _query_status_gcp(
     # GCP does not clean up preempted TPU VMs. We remove it ourselves.
     # TODO(wei-lin): handle multi-node cases.
     if use_tpu_vm and len(status_list) == 0:
-        logger.debug(f'Terminating preempted TPU VM cluster {cluster}')
-        backend = backends.CloudVmRayBackend()
-        handle = global_user_state.get_handle_from_cluster_name(cluster)
-        # Do not use refresh cluster status during teardown, as that will
-        # cause inifinite recursion by calling cluster status refresh
-        # again.
+        # Cleanup the preempted TPU directly with gcloud CLI.
```

Comment:
Can we add a comment of why we use CLI instead of using the `backend.teardown_no_lock` here? Something like: 
```
# We don't use the backend.teardown_no_lock because:
# 1. It is safe to call the CLI to terminate the cluster here as the cluster as none of the nodes is UP and the ray autoscaler should have been disabled.
# 2. ...
```

Reply:
If only part of the nodes are preempted, will this still work for cleaning up the clusters?

Reply:
Comment added!

> If only part of the nodes are preempted, will this still work for cleaning up the clusters?

This is a great point! thanks a lot for catching this. I've updated the code to fix this case by adding an argument `preempted_only` to `tpu_utils.terminate_tpu_vm_cluster_cmd`
https://github.com/skypilot-org/skypilot/pull/1185/files#diff-5d3b8c053073eef6241a9fc4439b8e97c680e73e6dbfe01a587eea2bf0000c4dR1642